### DISCUSSION TITLE
ARDESPOT problem model error

### DISCUSSION THREAD

**Discussion by Etwari3** [JuliaPOMDP/POMDPs.jl#484] [Debugging Help]
Hello,

I got this message running a continuous POMDP - "MethodError: Cannot `convert` an object of type Vector{Float64} to an object of type Int64".

Can you please help fix it?

Thank you.
[PipeCS_ARDESPOT_usingTOR_model2 - Jupyter Notebook.pdf](https://github.com/JuliaPOMDP/POMDPs.jl/files/11391993/PipeCS_ARDESPOT_usingTOR_model2.-.Jupyter.Notebook.pdf)

**Comment by WhiffleFish** [comment]
Hi @Etwari3, this error is likely occurring because your observation function is returning a single observation rather than a distribution.

The following observation function may get things to run as intended:

```julia
function POMDPs.observation(m::PipeCS, a::Int, sp::Float64)
    n = 15
    _s = round(Int, sp*100)
    return POMDPTools.Uniform(max(_s-n, 0):min(_s+n, 95))
end
```

I'm using POMDPTools for this, but you can use the older POMDPModelTools if need be.

**Reply by Etwari3** [reply]
Thank you.
